assets/js/base/components/product-name/index.tsx

```diff
@@ -10,10 +10,25 @@ import { AnchorHTMLAttributes, HTMLAttributes } from 'react';
  */
 import './style.scss';
 
-interface ProductNameProps extends AnchorHTMLAttributes< HTMLAnchorElement > {
+export interface ProductNameProps
+	extends AnchorHTMLAttributes< HTMLAnchorElement > {
+	/**
+	 * If `true` renders a `span` element instead of a link
+	 */
 	disabled?: boolean;
+	/**
+	 * The product name
+	 *
+	 * Note: can be an HTML string
+	 */
 	name: string;
```

Comment:
I love pedantic! And I love these kind of things. And this is something I generally want to bring up in teams, because consistency is key, I think. I tend to do alphabetical order (in my own project I even use the eslint rule for that), because I think it makes it easier to scan for people who don't know the code.

Though it sometimes creates unintuitive prop orders such as `height` before `width` and also you could have a number of props between these two.

I consider this an acceptable drawback, however.

Maybe alphabetical required, alphabetical optional could be a good ordering principle. I'll write a brief post about it and we can discuss, what say ye?

Reply:
Sounds good, leave it as is for now 👍